packages/falso/src/lib/core/core.ts

```diff
@@ -1,40 +1,116 @@
 import { random } from '../random';
+import { primitiveValueIsUnique } from './unique-validators';
 
 export interface FakeOptions {
   length?: number;
+  priority?: 'length' | 'unique';
   locale?: any | string[];
 }
 
 export type markRequired<Type, Key extends keyof Type> = Type & {
   [Property in Key]-?: Type[Property];
 };
 
+export interface FakeConfig<T> {
+  uniqueComparer: (
+    item: T,
+    items: T[],
+    comparisonKeys?: (keyof T)[]
+  ) => boolean | ((item: T, items: T[]) => boolean);
+  comparisonKeys?: (keyof T)[];
+}
+
 export type Return<T, O extends FakeOptions> = [O] extends [never]
   ? T
   : O['length'] extends number
   ? T[]
   : T;
 
-type FactoryFunction<T> = (i: number) => T;
+export type FactoryFunction<T> = (i: number) => T;
 
 export function fake<T, Options extends FakeOptions>(
   data: Readonly<T[]> | FactoryFunction<T>,
-  options?: Options
+  options?: Options,
+  config: FakeConfig<T> = {
+    uniqueComparer: primitiveValueIsUnique,
+  }
 ): Return<T, Options> {
-  const dataSource = Array.isArray(data)
-    ? () => randElement(data)
-    : (data as FactoryFunction<T>);
+  if (options?.length === 0) {
+    return [] as any;
+  }
+
+  if (Array.isArray(data)) {
+    return fakeFromArray(data, options) as any;
+  }
+
+  return fakeFromFunction(data as FactoryFunction<T>, config, options) as any;
+}
+
+export function fakeFromFunction<T, Options extends FakeOptions>(
+  data: FactoryFunction<T>,
+  config: FakeConfig<T>,
+  options?: Options
+): T | T[] {
+  if (!options?.length) {
+    return data(0);
+  }
+
+  const priority = options?.priority ?? 'length';
+
+  if (priority === 'length') {
+    return Array.from({ length: options.length }, (_, index) => data(0));
+  }
+
+  const items: T[] = [];
+
+  let attempts = 0;
+  const maxAttempts = options.length * 2;
+
+  while (items.length < options.length && attempts < maxAttempts) {
+    const item = data(0);
```

Comment:
I think it makes sense to pass the attempts as the function's argument.

Reply:
@shaharkazaz - How do you mean? To allow the user to decide the total number of attempts?

Reply:
You are passing 0 to that factory function, I think it makes more sense to pass the `attempts`, if the factory is using the index passed than you will increase the chances of getting the same result.
@NetanelBasal WDYT?

Reply:
Agree